cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -500,13 +617,155 @@ After .NET Core 2.0 it is possible to automatically generate and verify the anti
+
+DO NOT: Log sesnsitive data such as user's passwords.
+
+### Logging
+
+What Logs to Collect and more information about Logging can be found on this [cheat sheet](Logging_Cheat_Sheet.md).
+
+.NET Core come with a LoggerFactory, which is in Microsoft.Extensions.Logging. More information about ILogger can be found [here](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.ilogger).
+
+How to log all errors from the `Startup.cs`, so that anytime an error is thrown it will be logged.
+
+``` csharp
+public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+{
+	if (env.IsDevelopment())
+	{
+		_isDevelopment = true;
+		app.UseDeveloperExceptionPage();
+	}
+	
+	//Log all errors in the application
+	app.UseExceptionHandler(errorApp =>
+	{
+		errorApp.Run(async context =>
+		{
+		    var errorFeature = context.Features.Get<IExceptionHandlerFeature>();
+		    var exception = errorFeature.Error;
+		    
+		    Log.Error(String.Format("Stacktrace of error: {0}",exception.StackTrace.ToString()));
+		});
+	});
+
+        app.UseAuthentication();
+            app.UseMvc();
+        }
+}
+```
+
+e.g Injecting into the class constructor, which makes writing unit test simpler. It is recommended if instances of the class will be created using dependency injection (e.g. MVC controllers).  The belwo exaple shows logging of all unsucessful log in attempts.
+
+``` csharp
+public class AccountsController : Controller
+{
+        private ILogger _Logger;
+
+        public AccountsController( ILogger logger)
+        {
+            _Logger = logger;
+        }
+
+	[HttpPost]
+        [AllowAnonymous]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Login(LoginViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var result = await _signInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, lockoutOnFailure: false);
+                if (result.Succeeded)
+                {
+			//Log all successful log in attempts
+			Log.Information(String.Format("User: {0}, Successfully Logged in", model.Email));
+			//Code for successful login
+		}
+		else
+		{
+			//Log all incorrect log in attempts
+			Log.Information(String.Format("User: {0}, Incorrect Password", model.Email));
+		}
+	}
+	
+	...
+```
+
+Logging levels for ILogger are listed below, in order of high to low importance:
+
+### Monitoring
+
+Monitoring allow us to validate the performance and health of a running system through key performance indicators.
+
+In .NET a great option to add monitoring capabilities is [Application Insights](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-asp-net-core). 
+
+More information about Logging and Monitoring can be found [here](https://microsoft.github.io/code-with-engineering-playbook/Engineering/DevOpsLoggingDetailsCSharp.html).
+
+# OWASP 2013
+
+Below is vulnerability not discussed in OWASP 2017
```

Comment:
It miss the info link to the [Unvalidated Redirects and Forwards Cheat Sheet(https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.md) like for the LDAP injection.